Client/Frontend/Browser/Onboarding/OnboardingRewardsAgreementView.swift

```diff
@@ -0,0 +1,249 @@
+            descriptionLabel.accessibilityValue = nil
+            descriptionLabel.isAccessibilityElement = true
+        }
+        
+        init(theme: Theme) {
+            super.init(frame: .zero)
+            
+            applyTheme(theme)
+            mainStackView.tag = OnboardingViewAnimationID.details.rawValue
+            descriptionStackView.tag = OnboardingViewAnimationID.detailsContent.rawValue
+            imageView.tag = OnboardingViewAnimationID.background.rawValue
+            
+            addSubview(imageView)
+            addSubview(mainStackView)
+            mainStackView.snp.makeConstraints {
+                $0.leading.equalTo(self.safeArea.leading)
+                $0.trailing.equalTo(self.safeArea.trailing)
+                $0.bottom.equalTo(self.safeArea.bottom)
+            }
+            
+            descriptionView.addSubview(descriptionStackView)
+            descriptionStackView.snp.makeConstraints {
+                $0.edges.equalToSuperview().inset(UX.descriptionContentInset)
+            }
+            
+            mainStackView.addArrangedSubview(descriptionView)
+
+            [cancelButton, agreeButton, UIView.spacer(.horizontal, amount: 0)]
+                .forEach(buttonsStackView.addArrangedSubview(_:))
+            
+            [textStackView, buttonsStackView].forEach(descriptionStackView.addArrangedSubview(_:))
+            
+            descriptionCheckbox.addTarget(self, action: #selector(onTermsAccepted(_:)), for: .touchUpInside)
+        }
+        
+        func applyTheme(_ theme: Theme) {
+            descriptionView.backgroundColor = OnboardingViewController.colorForTheme(theme)
+            titleLabel.appearanceTextColor = theme.colors.tints.home
+            updateDescriptionLabel()
+            descriptionCheckbox.setImage(theme.isDark ? #imageLiteral(resourceName: "checkbox_off_dark") : #imageLiteral(resourceName: "checkbox_off"), for: .normal)
+        }
+        
+        override func layoutSubviews() {
+            super.layoutSubviews()
+            
+            let size = imageView.intrinsicContentSize
+            let scaleFactor = bounds.width / size.width
+            let newSize = CGSize(width: size.width * scaleFactor, height: size.height * scaleFactor)
+            
+            imageView.frame = CGRect(x: 0.0, y: UX.animationContentInset, width: newSize.width, height: newSize.height)
+        }
+        
+        @available(*, unavailable)
+        required init(coder: NSCoder) { fatalError() }
+        
+        @objc
+        private func onTermsAccepted(_ button: UIButton) {
+            button.isSelected.toggle()
+            
+            agreeButton.backgroundColor = button.isSelected ? BraveUX.BraveOrange : BraveUX.BraveOrange.withAlphaComponent(0.7)
+            agreeButton.isEnabled = button.isSelected
+        }
+        
+        override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
```

Comment:
Why not just override `pointInside` instead of hit test:

```
override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
  if UIEdgeInsetsInsetRect(bounds, UX.checkboxInsets)).contains(point) {
    return true
  }
  return super.point(inside: point, with: event)
}
```

Reply:
In case we are wondering. I HAVE to add it on the main view and not the button itself because the stackView is legit blocking the input to the button on the top and bottom (it clips it).. but NOT the sides due to the label next to it lol.

Reply:
Yeah stack view's a bitch for not accepting touches outside its bounds :P